ghc.nix

```
@@ -203,14 +206,19 @@ hspkgs.shellFor rec {
     export GHCPKG=$NIX_GHCPKG
     export HAPPY=${happy}/bin/happy
     export ALEX=${alex}/bin/alex
+    export CONFIGURE=configure
     ${lib.optionalString withEMSDK "export EMSDK=${emscripten}"}
     ${lib.optionalString withEMSDK "export EMSDK_LLVM=${emscripten}/bin/emscripten-llvm"}
     ${ # prevents sub word sized atomic operations not available issues
        # see: https://gitlab.haskell.org/ghc/ghc/-/wikis/javascript-backend/building#configure-fails-with-sub-word-sized-atomic-operations-not-available
       lib.optionalString withEMSDK ''
+      unset CC
+      CONFIGURE="emconfigure ./configure"
       cp -Lr ${emscripten}/share/emscripten/cache .emscripten_cache
       chmod u+rwX -R .emscripten_cache
       export EM_CACHE=.emscripten_cache
```

Ah we should replace these three lines because they are a hack. The problem is that this code never cleans up the `.emscripten_cache` so the file system gets littered with duplicates after a while.

This should work:

```suggestion
export EM_CACHE=$(mktemp -d)
```

For the record that comes from the preconfigure phase of [ghcjs in Haskell.nix](https://github.com/input-output-hk/haskell.nix/blob/bc3f8b660a3b40f52139f59213652e083e6d2418/compiler/ghc/default.nix#L351).